front/src/modules/analytics/hooks/useTrackPageView.ts

```diff
@@ -1,17 +1,26 @@
 import { useEffect } from 'react';
 import { useLocation } from 'react-router-dom';
 
+import usePrevious from '@/utils/hooks/usePrevious';
+
 import { useEventTracker } from './useEventTracker';
 
 export function useTrackPageView() {
   const location = useLocation();
+  const previousLocation = usePrevious(location);
```

Comment:
Is it possible with a recoil state ?

Reply:
I guess so, but I think it's a bit overkill to use in that case as we don't need to have a state scoped on the all app